index.js

```diff
@@ -30,60 +56,85 @@ const checkVersion = (engineName, command) => {
     // eslint-disable-next-line global-require, import/no-dynamic-require
     pkg = require(pkgJsonPath);
   } catch (e) {
-    console.log(`Unable to find ${pkgJsonPath}!  😱`);
-    console.log('');
+    console.log(`Unable to find ${pkgJsonPath}!  😱${EOL}`);
     console.log(
-      'Please ensure that this script is executed in the same directory.',
+      `Please ensure that this script is executed in the same directory.${EOL}`,
     );
-    console.log('');
     process.exit(1);
   }
 
-  let expected = null;
-
-  try {
-    expected = pkg.engines[engineName];
-  } catch (e) {
+  if (!pkg.engines || !pkg.engines[engineName]) {
     console.log(
-      `There is no engine named ${engineName} specified in package.json!  😱`,
+      `There is no engine named "${engineName}" specified in package.json!  😱${EOL}`,
     );
-    console.log('');
     process.exit(1);
   }
 
-  let version = null;
+  const expected = pkg.engines[engineName];
+
+  let globalVersion = null;
+  let localVersion = null;
+  let usedVersion = null;
+  let globalVersionValid = null;
+  let localVersionValid = null;
 
   try {
-    version = execSync(command)
+    const localBinFolder = execSync('npm bin')
       .toString()
-      .replace('v', '')
       .trim();
+
+    const hasLocalVersion = existsSync(
+      `${localBinFolder}${path.sep}${engineName}`,
+    );
+
+    globalVersion = getVersion({ command, localBinFolder });
+    globalVersionValid = semver.satisfies(globalVersion, expected);
+
+    console.log(`Expected ${engineName} version:\t${expected}`);
+    console.log(
+      `╰─ Global ${engineName} version:\t${globalVersion}\t${
+        globalVersionValid ? '✅️' : '❌️'
+      }`,
+    );
+
+    if (hasLocalVersion) {
+      localVersion = getVersion({ command, localBinFolder, global: false });
+      localVersionValid = semver.satisfies(localVersion, expected);
+      console.log(
+        `╰─ Local ${engineName} version:\t${localVersion}\t${
+          localVersionValid ? '✅️' : '❌️'
+        } (from ${localBinFolder})`,
+      );
+    }
+
+    if (userInput.ignoreLocalBin) {
+      usedVersion = globalVersion;
+    } else {
+      usedVersion = localVersion || globalVersion;
+    }
+
+    console.log(`╰─ (using: ${usedVersion})${EOL}`);
```

Comment:
I'm not the biggest fan of the verbose output now ...
```
Checking node & npm versions...

Expected node version:  ^8.12.0
╰─ Global node version: 8.12.0  ✅️
╰─ (using: 8.12.0)

Expected npm version:   ^6.4.1
╰─ Global npm version:  6.4.1   ✅️
╰─ (using: 6.4.1)


All good.  👍
```

Fair enough if something wen't wrong ...

Reply:
I thought of adding a `--verbose` param, but most of the output is printed _before_ the script knows whether the check has passed or failed - so, to print the output only in case of errors some refactoring would be needed. Not sure it's worth it.

Personally, I like the informative output (because I like to see what I'm running and it trains me to see immediately when something changes), but I understand it comes down to personal preference.

Reply:
How about we keep the information short when we exit with a status of `0` and have verbose output when we exit with a non-zero exit code?

Reply:
I'm not the biggest fan of this too as most of the time this check pass (or I hope so 😅 ) and the verbose output is a bit distracting and harder to read than the previous version.

This is small enough though so wouldn't add that much noise.

Reply:
Yeah push that 👍

Reply:
Done. Default output now is:

```
Checking node & npm versions...


All good.  👍
```

And, in case of errors:

```
Checking node & npm versions...

Expected npm version to match ^5.6.0, but got 5.0.2.  😱

Please follow Skyscanner's node environment guide (see https://github.com/Skyscanner/ensure-node-env/blob/master/README.md#guide).
```

With the `--verbose` flag, it shows the information in the original screenshot.

Thanks for the feedback! 😉

Reply:
Build failed 😇